### PR TITLE
BACKLOG-10372 : add input component in design system

### PR DESCRIPTION
**Pull request by dgriffon**

### CODE REVIEW

File: src/javascript/DesignSystem/Input/Input.jsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+                className={`${inputAdornedStart} ${!readOnly && focus ? inputAdornedStartFocus : ''} ${!readOnly && error ? inputAdornedStartError : ''}`}
+                position="start"
+            >
+                {icon}
+            </InputAdornment>}
+            endAdornment={interactive && <InputAdornment position="end">{interactive}</InputAdornment>}
+            value={value}
+            onBlur={handleBlur}
+            onFocus={handleFocus}
+            onChange={onChange}
```
Comment (FBerthelot) [inline src/javascript/DesignSystem/Input/Input.jsx]: Tu ne fais aucun traitement dessus, autant le `{...other}` gérer ces props

File: src/javascript/DesignSystem/Input/index.js [hunk 2]
```diff
@@ -0,0 +1 @@
+export {Input} from './Input';
```
Comment (FBerthelot) [inline src/javascript/DesignSystem/Input/index.js]:
```suggestion
export * from './Input';
```

File: src/javascript/DesignSystem/Input/Input.jsx (hunk 1 above)
Comment (dgriffon) [reply src/javascript/DesignSystem/Input/Input.jsx]: onBlur and onFocus are used by the focus state handling of the component.
As we want to expose our component (Material UI should be hidden) we need define onChange and value as prop used by some of our components.

Comment (FBerthelot) [reply src/javascript/DesignSystem/Input/Input.jsx]: They have to be in the prop-types for sure, but what is the gain to do ` onChange={onChange}` vs `{...other}` ?

File: src/javascript/DesignSystem/Input/Input.jsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+    const {readOnly: readOnlyClass, inputDisabled, inputAdornedStart, inputAdornedStartFocus, inputAdornedStartError, ...containerClasses} = classes;
+
+    return (
+        <InputBase
+            classes={containerClasses}
+            className={`${readOnly ? readOnlyClass : ''} ${disabled ? inputDisabled : ''}`}
+            disabled={disabled}
+            error={error}
+            readOnly={readOnly}
```
Comment (FBerthelot) [inline src/javascript/DesignSystem/Input/Input.jsx]:
```suggestion
```

File: src/javascript/DesignSystem/Input/Input.jsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    const {icon, interactive} = variant;
+
+    const {readOnly: readOnlyClass, inputDisabled, inputAdornedStart, inputAdornedStartFocus, inputAdornedStartError, ...containerClasses} = classes;
+
+    return (
+        <InputBase
+            classes={containerClasses}
+            className={`${readOnly ? readOnlyClass : ''} ${disabled ? inputDisabled : ''}`}
+            disabled={disabled}
+            error={error}
```
Comment (FBerthelot) [inline src/javascript/DesignSystem/Input/Input.jsx]:
```suggestion
```

File: src/javascript/DesignSystem/Input/Input.jsx [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+    const {icon, interactive} = variant;
+
+    const {readOnly: readOnlyClass, inputDisabled, inputAdornedStart, inputAdornedStartFocus, inputAdornedStartError, ...containerClasses} = classes;
+
+    return (
+        <InputBase
+            classes={containerClasses}
+            className={`${readOnly ? readOnlyClass : ''} ${disabled ? inputDisabled : ''}`}
+            disabled={disabled}
```
Comment (FBerthelot) [inline src/javascript/DesignSystem/Input/Input.jsx]:
```suggestion
```

File: src/javascript/DesignSystem/Input/Input.jsx [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+        inputAdornedStartFocus: {
+            color: theme.palette.brand.alpha
+        },
+        inputAdornedStartError: {
+            color: theme.palette.support.alpha
+        }
+    };
+};
+
+export const InputCmp = ({classes, disabled, error, inputProps, onBlur, onFocus, readOnly, variant, ...others}) => {
```
Comment (FBerthelot) [inline src/javascript/DesignSystem/Input/Input.jsx]:
```suggestion
export const InputCmp = ({classes, inputProps, onBlur, onFocus, readOnly, variant, ...others}) => {
```

**Pull request merged**